src/sagas/wallet.js

```diff
@@ -261,10 +264,21 @@ export function* startWallet(action) {
   // another saga (using the `fork` effect), it will remain active until all
   // the forks are terminated. You can read more details at
   // https://redux-saga.js.org/docs/advanced/ForkModel
-  // So, if a new START_WALLET_REQUESTED action is dispatched, we need to cleanup
-  // all attached forks (that will cause the event listeners to be cleaned).
-  yield take(types.START_WALLET_REQUESTED);
+  // So, if a new START_WALLET_REQUESTED action is dispatched or a RELOAD_WALLET_REQUESTED
+  // is dispatched, we need to cleanup all attached forks (that will cause the event
+  // listeners to be cleaned).
+  const { reload } = yield race({
+    start: take(types.START_WALLET_REQUESTED),
+    reload: take(types.RELOAD_WALLET_REQUESTED),
+  });
+
+  // We need to cancel threads on both reload and start
   yield cancel(threads);
+
+  if (reload) {
+    // Yield the same action again to reload the wallet
+    yield put(action);
```

Comment:
Why do we need to reload again here? Haven't we done that above in line 228?

Reply:
Yes, but the idea here is to capture the `RELOAD_WALLET_REQUESTED` (that might be dispatched from anywhere in the code) and yield the same action that started the current `startWallet` saga

An example is on line 396